tt-train/README.md

```diff
@@ -0,0 +1,75 @@
+# tt-train: CPP ML training framework
+
+## Overview
+This repository contains a high-performance training framework developed in C++ designed to efficiently leverage the computational capabilities of Tenstorrent hardware. The framework is optimized to accelerate model training tasks, providing a seamless interface for training deep learning models on Tenstorrent's advanced hardware architecture.
+
+# Install
+1. Initialize and update submodules
+```
+git submodule update --init --recursive
+```
+2. Install CMake 3.30
+3. Run setup script to configure env variables, direnv, clang-tidy and clang-format.
+```
+source ./init_repo.sh
+```
+4. Navigate to `tt-metal` folder and follow repository instructions to build it
+
+
+# Building the project:
+You have two options for building the project:
+
+## 1. VSCode
+* Install the [CMake](https://marketplace.visualstudio.com/items?itemName=twxs.cmake) and [direnv](https://marketplace.visualstudio.com/items?itemName=mkhl.direnv) extensions for VSCode.
+* Use UI to build all targets.
+
+## 2. Terminal
+### Debug
```

Comment:
Should not share the same bin dir across different build types.  Either use a multi-config generator like so:

```bash
cmake -B build -G "Ninja Multi-Config"
cmake --build build --config Debug
cmake --build build --config Release
```
Or use different build dirs
```bash
cmake -B build/debug -G "Ninja" -DCMAKE_BUILD_TYPE=Debug
cmake --build build/debug
cmake -B build/release -G "Ninja" -DCMAKE_BUILD_TYPE=Release
cmake --build build/release
```